stellar_rust_sdk_derive/src/lib.rs

```diff
@@ -7,9 +7,10 @@ use syn::{parse_macro_input, DeriveInput};
 pub fn pagination_macro(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
     let struct_name = &input.ident;
-
+    let (impl_generics, type_generics, where_clause) = input.generics.split_for_impl();
+    
     let expanded = quote! {
-        impl Paginatable for #struct_name {
+        impl #impl_generics Paginatable for #struct_name #type_generics #where_clause  {
```

Comment:
Kunnen we het hier een keer over hebben? Ik zeg niet dat het fout is maar ik wil het wat beter begrijpen voordat ik het approve.